casper/src/main/scala/coop/rchain/casper/engine/Running.scala

```diff
@@ -77,21 +81,22 @@ object Running {
     }
 
     def tryRerequest(hash: BlockHash, requested: Requested): F[(BlockHash, Option[Requested])] =
-      if (requested.waitingList.nonEmpty) {
-        val nextPeer = requested.waitingList(0)
-        def modifiedRequested(ts: Long) = requested.copy(
-          timestamp = ts,
-          waitingList = requested.waitingList.tail,
-          peers = requested.peers + nextPeer
-        )
-        for {
-          _  <- requestForBlock(nextPeer, hash)
-          ts <- Time[F].currentMillis
-        } yield ((hash -> Option(modifiedRequested(ts))))
-      } else {
-        val warnMessage = s"Could not retrieve requested block ${PrettyPrinter.buildString(hash)}. " +
-          "Removing the request from the requested blocks list. Casper will have to re-request the block."
-        Log[F].warn(warnMessage).as((hash -> none[Requested]))
+      requested.waitingList match {
+        case nextPeer :: waitingListTail =>
+          def modifiedRequested(ts: Long) = requested.copy(
```

Comment:
I am a bit against defining a nested function so deep into the outer function's code. Not a deal-breaker by any means, but you might reconsider moving it closer to the top

Reply:
I am also not sure if it should be a function - it's quite obvious what the code does, there is no extra logic that would warrant an encapsulation IMO. Maybe reconsider inlining it?